src/plugins/console/public/application/models/legacy_core_editor/legacy_core_editor.ts

```diff
@@ -414,4 +414,42 @@ export class LegacyCoreEditor implements CoreEditor {
   destroy() {
     this.editor.destroy();
   }
+
+  autoIndent(reqRange: Range) {
+    const session = this.editor.getSession();
+    const mode = session.getMode();
+    const startRow = reqRange.start.lineNumber;
+    const endRow = reqRange.end.lineNumber;
+
+    let prevLineState = '';
+    let prevLine = '';
+    let lineIndent = '';
+    let line;
+    let currIndent;
+    let range;
+    const tab = session.getTabString();
+
+    for (let row = startRow; row <= endRow; row++) {
+      if (row > 0) {
+        prevLineState = session.getState(row - 1);
+        prevLine = session.getLine(row - 1);
+        lineIndent = mode.getNextLineIndent(prevLineState, prevLine, tab);
+      }
+
+      line = session.getLine(row);
+      // @ts-ignore we have access to $getMode
```

Comment:
Ace API provides this method for text mode, since we extend our input mode from text mode we can access this method. However, the [brace](https://github.com/thlorenz/brace) package we are using does not provide a type definition for this method and ts complains with `TS2339: Property '$getIndent' does not exist on type 'TextMode'`. `eslint-disable-next-line` doesn't seem to work in this case. I left some comments on this line to explain why we used `@ts-ignore` annotation.